project/src/components/catalog-films-list/catalog-films-list.tsx

```diff
@@ -1,16 +1,23 @@
 import FilmCard from '../film-card/film-card';
-import {FilmType} from '../../types/film';
-
+import { FilmType } from '../../types/film';
+import { useState } from 'react';
 
 type CatalogFilmsListProps = {
-  filmCardsData: FilmType[]
-}
+  filmCardsData: FilmType[];
+};
+
+function CatalogFilmsList({ filmCardsData }: CatalogFilmsListProps): JSX.Element {
+  const [activeCard, setActiveCard] = useState(0);
 
+  const updateActiveCard = (value: number): void => {
+    setActiveCard(value);
+  };
 
-function CatalogFilmsList({filmCardsData}: CatalogFilmsListProps): JSX.Element {
   return (
     <div className="catalog__films-list">
-      {filmCardsData.map((filmData) => <FilmCard filmData={filmData} key={filmData.id} />)}
+      {filmCardsData.map((filmData) => (
+        <FilmCard filmData={filmData} key={filmData.id} onActive={updateActiveCard} />
```

Comment:
В реакте такие обработчики принято называться handle + что то. А вот пропс который спускаем в ребенка on + что то 
```suggestion
        <FilmCard filmData={filmData} key={filmData.id} onActive={handleHover(Card)} />
```